isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/services.adoc

```diff
@@ -0,0 +1,247 @@
+Queueing wird ausschließlich in der internen Servicekommunikation eingesetzt.
+
+JMS-Nachrichten bestehen aus Header, Properties und einem Body.
+Die Properties unterteilen sich noch einmal in applikationsspezifische Properties, die nur für Publisher und Subscriber Bedeutung haben, sowie provider-spezifische und Standard-Properties, die zur Verarbeitung der JMS-Nachrichten durch den Message-Broker gedacht sind.
+
+[[jms-message-aufbau]]
+.Aufbau einer JMS-Nachricht
+image::software-technisch/jms-message-aufbau.dn.svg[]
+
+Applikationsspezifische Properties enthalten Metadaten.
+Der Body enthält Nutzdaten.
+Nutzdaten werden im XML-Format übertragen und mittels XSD spezifiziert.
+
+Diese Vorgabe steht vollständig in Einklang mit der JMS-Spezifikation.
+Für die Übertragung von Nutzdaten sieht die JMS-Spezifikation fünf Formate vor.
+Die Architekturvorgabe sieht die alleinige Nutzung der Ausprägung `TextMessage` vor, die Nutzdaten als Zeichenkette erwartet.
+
+[NOTE]
+====
+Weitere Details zu JMS-Nachrichten finden sich in der JMS-Spezifikation im Kapitel https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-model[3. Jakarta Messaging message model].
+Besonders relevant für die Referenzarchitektur sind die Abschnitte https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-messages[3.3. Jakarta Messaging messages] sowie https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-body[3.11. Jakarta Messaging message body].
+====
+
+== Kommunikation mit externen Systemen
+
+Die Kommunikation mit externen Systemen basiert auf Web-Services.
+Wird ein Service von einem externen System angeboten, wird er als *externer Service* bezeichnet.
+Hierbei muss man zwischen zwei Fällen unterscheiden:
+
+*Aufruf von Services der Systemlandschaft*: Durch die Systemlandschaft wird externen Systemen die Schnittstelle eines Backends in Form eines Web-Services zur Verfügung gestellt.
+Hierbei definiert das Backend selbst keinen Web-Service.
+Vielmehr definiert das Backend, wie bei der internen Kommunikation auch, eine Schnittstelle.
+Diese Schnittstelle wird dann durch ein eigenständiges IT-System als Web-Service exportiert.
+Dieses IT-System wird als *Service-Provider* bezeichnet.
+Für jede Schnittstelle, die als Web-Services exportiert werden soll, muss ein eigener Service-Provider definiert werden.
+
+*Nutzung von externen Services*: Ähnlich wie im vorigen Fall ruft das interne IT-System den externen Service nicht direkt auf.
+Es ruft ein eigenständiges IT-System auf, welches den externen Service als Schnittstelle in die Systemlandschaft importiert.
+Dieses IT-System wird als *Service-Consumer* bezeichnet.
+Das interne IT-System ruft dann lediglich die Schnittstelle des Service-Consumers auf.
+Für das interne IT-System ist dieser Aufruf nicht von einem Aufruf zu einem anderen internen IT-System zu unterscheiden.
```

Comment:
Verstehe ich das richtig: der Service-Consumer (im SGW) wird genauso angesprochen wie Systemlandschaft-interne Services: d.h. 
- wir nutzen zum Service-Aufruf in *beiden* Fällen eine REST API? (weil Sys-Landschaft-intern REST angesagt ist)
- oder eine WebService-API? (weil Sys-Landschaft-extern WebServices bereitgestellt werden)

Oder ist hier gemeint: 
- man greift in einer Anwendung auf Java-Interfaces, die von Adapter-Klassen implementiert sind.
- d.h. wenn man nur auf die Interfaces schaut, sieht man nicht, ob interne oder externe Kommunikation stattfindet
  wenn man allerdings auf die Implementierung dieser Interfaces schaut, sieht man es sehr wohl

Reply:
SGWs "übersetzen" die externen Serviceaufrufe so, dass Backends keinen Unterschied zu internen Serviceaufrufen bemerken. Das bedeutete oft einen Protokollwechsel und die Transformation der übermittelten Daten von einem Format ins andere. Auch das ist Input für ein Folgeticket, den Aufbau und die Rolle von SGWs ausführlicher zu beschreiben.